python/ray/experimental/workflow/execution.py

```diff
@@ -38,28 +59,23 @@ def run(entry_workflow: Workflow,
     if workflow_id is None:
         # Workflow ID format: {Entry workflow UUID}.{Unix time to nanoseconds}
         workflow_id = f"{entry_workflow.id}.{time.time():.9f}"
-    if isinstance(storage, str):
-        set_global_storage(create_storage(storage))
-    elif isinstance(storage, Storage):
-        set_global_storage(storage)
-    elif storage is not None:
-        raise TypeError("'storage' should be None, str, or Storage type.")
-    storage_url = get_global_storage().storage_url
+    store = _get_storage(storage)
     logger.info(f"Workflow job created. [id=\"{workflow_id}\", storage_url="
-                f"\"{storage_url}\"].")
-    try:
-        workflow_context.init_workflow_step_context(workflow_id, storage_url)
-        commit_step(entry_workflow)
-        actor = get_or_create_management_actor()
-        # NOTE: It is important to 'ray.get' the returned output. This
-        # ensures caller of 'run()' holds the reference to the workflow
-        # result. Otherwise if the actor removes the reference of the
-        # workflow output, the caller may fail to resolve the result.
-        output = ray.get(actor.run_or_resume.remote(workflow_id, storage_url))
-        direct_output = flatten_workflow_output(workflow_id, output)
-    finally:
-        workflow_context.set_workflow_step_context(None)
-    return direct_output
+                f"\"{store.storage_url}\"].")
+
+    # checkpoint the workflow
+    ws = workflow_storage.WorkflowStorage(workflow_id, store)
+    ws.save_subworkflow(entry_workflow)
+    ws.save_step_output("", entry_workflow, None)
+
+    actor = get_or_create_management_actor()
```

Comment:
Maybe you missed the comment last time, bud could you update the name here?
actor -> workflow_manager

Reply:
hmmm, seems this is a different one. let me fix it